### PR TITLE
 Personal data clear (Task #11954)

### PR DESCRIPTION
**Pull request by eliamariutti**
If a module has personal data marked in the 'fields.json', in the view will show a new "Clear personal data" button.
Once clicked, from the record will be erased all the fields marked as personal after a configured period.
The shell script will take cake of pick all the expired records of a selected module (or all the them), and erase the data.

### CODE REVIEW

File: config/Migrations/20190801083855_ScheduledPersonalData.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+     * Change Method.
+     *
+     * More information on this method is available here:
+     * http://docs.phinx.org/en/latest/migrations.html#the-change-method
+     * @return void
+     */
+    public function change()
+    {
+        $table = $this->table('scheduled_personal_data', ['id' => false, 'primary_key' => ['id']]);
```
Comment (acrias) [inline config/Migrations/20190801083855_ScheduledPersonalData.php]: Is this needed as we have below the addPrimaryKey?

Comment (eliamariutti) [reply config/Migrations/20190801083855_ScheduledPersonalData.php]: @acrias Actually this is needed due phinx add an incremental id field (https://book.cakephp.org/3.0/en/phinx/migrations.html#creating-a-table), but I will drop the `$table->addPrimaryKey(['id']);`

File: src/Shell/ClearPersonalDataShell.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+     * @return mixed[]
+     */
+    private function getPersonalField(string $moduleName) : array
+    {
+        $config = (array)(new ModuleConfig(ConfigType::FIELDS(), $moduleName))->parseToArray();
+        $fields = array_filter($config, function ($v) {
+            return in_array('personal', (array)array_keys($v));
+        });
+
+        return array_map(function () {
```
Comment (georgeconstantinou) [inline src/Shell/ClearPersonalDataShell.php]: @eliamariutti is this returning key-value pairs, with personal fields as keys and empty strings as value?

Comment (eliamariutti) [reply src/Shell/ClearPersonalDataShell.php]: yes, exactly!

**Pull request closed without merging**